### PR TITLE
Feature/genericize event tracking

### PR DESCRIPTION
**Pull request by thundercat1**

### CODE REVIEW

**Comment by scb-michael-nguyen** [comment]
Checking my understanding: you're adding an interface to allow passing in any "tracking function", which in this case just gets passed the object we'd originally pass to react GA. So Dealer locator instances will require a tracking function from whatever project they're in.

**Comment by thundercat1** [comment]
> n

Exactly -- the project using dealer locator will be able to implement tracking using whatever method they want (or pass in a no-op and don't track events at all). In this case I'll have a code change in scb-static to send the events to the GTM dataLayer, but more generally I want to be able to change it to anything I want in the future without having to make a PR to dealer-locator.

**Review by scb-michael-nguyen — APPROVED** [review]

File: src/dealer-details.js [hunk 1]
```diff
@@ -21,15 +20,15 @@ const createDealerDirectionsURL = (dealer) => {
   return `https://www.google.com/maps/dir/?api=1&destination=${dealerLocationQuery}`;
 };
 
-const dealerConversionEvent = (dealer, action) => {
-  ReactGA.event({
+const dealerConversionEvent = (dealer, action, trackEvent) => {
+  trackEvent({
     category: "Dealer Locator",
     action: action,
     label: _get(dealer, "name"),
```
Comment (apsummers) [inline src/dealer-details.js]: This seems strange to me. The ReactGA API isn't necessarily the same as any given tracking library, but this PR treats any function we pass into it as `ReactGA.event` flavored.

Comment (thundercat1) [reply src/dealer-details.js]: That's true, we could pass a more generic event object up and format it for GA in the main project, and labeling category "Dealer Locator" is totally useless.

I think we do need to separate out the action type & a descriptive label, `{action: foo, label: bar}` uses the GA convention but I can't really think of any better way to name the object keys. Any other proposals?

Comment (apsummers) [reply src/dealer-details.js]: I'd say a prop for each of the on-click events that we think should be able to execute some kind of callback. `onDealerWebsiteButtonClick`, `onDealerPhoneClick`, etc. That way we can pass in whatever we feel like (`ReactGA.event({ action: "foo", label: "bar" })` or `GTM.doSomething("foo", "bar")` or `myCoolCallback(42)`) without having any implicit dependencies (the API surface) and we can chain whatever we want, or not.

Comment (thundercat1) [reply src/dealer-details.js]: > I'd say a prop for each of the on-click events that we think should be able to execute some kind of callback. `onDealerWebsiteButtonClick`, `onDealerPhoneClick`, etc. That way we can pass in whatever we feel like (`ReactGA.event({ action: "foo", label: "bar" })` or `GTM.doSomething("foo", "bar")` or `myCoolCallback(42)`) without having any implicit dependencies (the API surface) and we can chain whatever we want, or not.

Added & tested with new props for these callbacks! It's a bit more code but should be really flexible

**Review by apsummers — APPROVED** [review]

**Pull request merged**